joint_teapot/workers/gitea.py

```diff
@@ -375,6 +376,32 @@ def archive_all_repos(self) -> None:
                 self.org_name, repo.name, body={"archived": True}
             )
 
+    def get_all_teams(
+        self,
+    ) -> List[StudentGroup]:
```

Comment:
No need for a separate dataclass, just return a dict {team_name: members}

Reply:
I was thinking this is a basic and common model we will use often, so perhaps defining a dataclass will benefit us in the long run. Initially I did a dict here, but I found it annoying to work with since we don't have autocompletion for dictionaries.. If there's a good reason I can switch back though.

Reply:
I'm not aware of that, and I was thinking about one dataclass as a unified interface that encompasses most of the info we need to work with and bridges different representations. Since student group is a very common model I decided to write a separate dataclass for it. Perhaps it's not easy to design such an interface, but it would make things clearer. Otherwise other devs would need to spend some time trying to figure out what the return values mean and how they are structured, or we would need to write extensive comments.

Reply:
student group data structure is not complicated, to keep consistency among the whole project, you need to do lots of extra work to create models for other return values and it makes things much more complicated. i do not think other devs need time to understand such a simple data structure.